### ISSUE TITLE
Vault 1.15.0 rateLimitExceeded with GCS

### ISSUE THREAD

**Issue by Lord-Y** [hashicorp/vault#23635]
Hello guys, since I migrated my vault cluster from version `1.14.4` to `1.15.0` I started to have rate limit issues with both `terraform` and `ansible` executions.

**Describe the bug**
Here is an example output with terraform:
```
│ Error: error reading from Vault: Error making API request.
│
│ URL: GET https://REDACTED:8200/v1/sys/internal/ui/mounts/secret/REDACTED
│ Code: 500. Errors:
│
│ * error performing token check: failed to read entry: 1 error occurred:
│       * error closing connection: googleapi: Error 429: The object REDACTED/sys/token/id/REDACTED exceeded the rate limit for object mutation operations (create, update, and delete). Please reduce your request rate. See https://cloud.google.com/storage/docs/gcs429., rateLimitExceeded
```

**Environment:**
* Vault Server Version (retrieve with `vault status`):
```
vault status
Key                      Value
---                      -----
Recovery Seal Type       shamir
Initialized              true
Sealed                   false
Total Recovery Shares    5
Threshold                3
Version                  1.15.0
Build Date               2023-09-22T16:53:10Z
Storage Type             gcs
Cluster Name             REDACTED
Cluster ID               REDACTED
HA Enabled               true
HA Cluster               https://REDACTED:8201
HA Mode                  active
Active Since             2023-10-12T14:23:14.976291598Z
```

* Vault CLI Version (retrieve with `vault version`):
* Server Operating System/Architecture:
```
lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 11 (bullseye)
Release:        11
Codename:       bullseye
```

Vault server leader log:
```
2023-10-12T14:56:16.615Z [ERROR] core: failed to construct ACL:
  error=
  | failed to get policy: failed to read policy: 1 error occurred:
  | \t* error closing connection: googleapi: Error 429: The object REDACTED/sys/policy/default exceeded the rate limit for object mutation operations (create, update, and delete). Please reduce your request rate. See https://cloud.google.com/storage/docs/gcs429., rateLimitExceeded
  |

  error=
  | failed to get policy: failed to read policy: 1 error occurred:
  | \t* error closing connection: googleapi: Error 429: The object REDACTED/sys/policy/REDACTED exceeded the rate limit for object mutation operations (create, update, and delete). Please reduce your request rate. See https://cloud.google.com/storage/docs/gcs429., rateLimitExceeded
  |
```

I didn't have any quota enabled before so I was like let set one:

```
vault write sys/quotas/rate-limit/global-rate rate=1000000000 interval=2m
```

```
vault read sys/quotas/rate-limit/global-rate
Key               Value
---               -----
block_interval    0
inheritable       true
interval          120
name              global-rate
path              n/a
rate              1000000000
role              n/a
type              rate-limit
```

Quota config:
```
vault read sys/quotas/config
Key                                   Value
---                                   -----
enable_rate_limit_audit_logging       false
enable_rate_limit_response_headers    false
rate_limit_exempt_paths               [sys/generate-recovery-token/attempt sys/generate-recovery-token/update sys/generate-root/attempt sys/generate-root/update sys/health sys/seal-status sys/unseal]
```

Even by setting a huge quota I'm still hitting this issue.
Before migrating, I never got this issue before. Do you guys have any ideas how to fix it?

**Comment by Lord-Y** [comment]
@divyaac I'm not sure why you closed the issue so quickly without waiting my feedback but anyway, it's okay.
So I removed the quota that I created, and then rollback vault to previous version `1.14.4` and everything is working perfecly just like before the upgrade to version `1.15.0`.
I have many GCS buckets and only the one related to `vault` was having this issue.
I hope this issue will help others.

Thx.
